### PR TITLE
Add table aws_cloudfront_origin_request_policy. Closes #424

### LINKED ISSUE #424: Add table aws_cloudfront_origin_request_policy
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

### PR DESCRIPTION
**Pull request by anisadas**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/aws_cloudfront_origin_request_policy []

PRETEST: tests/aws_cloudfront_origin_request_policy

TEST: tests/aws_cloudfront_origin_request_policy
Running terraform
data.aws_partition.current: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_region.alternate: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_cloudfront_origin_request_policy.named_test_resource: Creating...
aws_cloudfront_origin_request_policy.named_test_resource: Creation complete after 3s [id=09ba5b54-ad62-416d-ad5a-d76ae146562f]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = 986325076436
aws_partition = aws
e_tag = E23ZP02F085DFQ
resource_aka = arn:aws:cloudfront::986325076436:origin-request-policy/09ba5b54-ad62-416d-ad5a-d76ae146562f
resource_id = 09ba5b54-ad62-416d-ad5a-d76ae146562f
resource_name = turbottest26746

Running SQL query: test-get-query.sql
[
  {
    "account_id": "986325076436",
    "comment": "turbottest26746",
    "e_tag": "E23ZP02F085DFQ",
    "id": "09ba5b54-ad62-416d-ad5a-d76ae146562f",
    "title": "turbottest26746"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "account_id": "986325076436",
    "comment": "turbottest26746",
    "e_tag": "E23ZP02F085DFQ",
    "id": "09ba5b54-ad62-416d-ad5a-d76ae146562f",
    "name": "turbottest26746",
    "title": "turbottest26746"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:cloudfront::986325076436:origin-request-policy/09ba5b54-ad62-416d-ad5a-d76ae146562f"
    ],
    "title": "turbottest26746"
  }
]
✔ PASSED

POSTTEST: tests/aws_cloudfront_origin_request_policy

TEARDOWN: tests/aws_cloudfront_origin_request_policy

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
select
  name,
  id,
  comment,
  e_tag,
  last_modified_time,
  type
from
  aws_cloudfront_origin_request_policy;

+-----------------------------------------------------+--------------------------------------+------------------------------------------------------------+----------------+-------------------
| name                                                | id                                   | comment                                                    | e_tag          | last_modified_time
+-----------------------------------------------------+--------------------------------------+------------------------------------------------------------+----------------+-------------------
| Managed-UserAgentRefererHeaders                     | acba4595-bd28-49b8-b9fe-13317c0390fa | Policy to forward user-agent and referer headers to origin | E23ZP02F085DFQ | 1970-01-01 00:00:0
| Managed-AllViewer                                   | 216adef6-5c7f-47e4-b989-5492eafa07d3 | Policy with forward all parameters in viewer requests      | E23ZP02F085DFQ | 1970-01-01 00:00:0
| Managed-CORS-S3Origin                               | 88a5eaf4-2fd4-4709-b370-b4c650ea3fcf | Policy for S3 origin with CORS                             | E23ZP02F085DFQ | 1970-01-01 00:00:0
| Managed-Elemental-MediaTailor-PersonalizedManifests | 775133bc-15f2-49f9-abea-afb2e0bf67d2 | Policy for Elemental MediaTailor Origin                    | E23ZP02F085DFQ | 1970-01-01 00:00:0
| Managed-CORS-CustomOrigin                           | 59781a5b-3903-41f3-afcb-af62929ccde1 | Policy for custom origin with CORS                         | E23ZP02F085DFQ | 1970-01-01 00:00:0
+-----------------------------------------------------+--------------------------------------+------------------------------------------------------------+----------------+-------------------

select
  name,
  id,
  headers_config ->> 'HeaderBehavior' as header_behavior,
  headers_config ->> 'Headers' as headers
from
  aws_cloudfront_origin_request_policy;
  
+-----------------------------------------------------+--------------------------------------+-----------------+-------------------------------------------------------------------------------
| name                                                | id                                   | header_behavior | headers                                                                       
+-----------------------------------------------------+--------------------------------------+-----------------+-------------------------------------------------------------------------------
| Managed-CORS-S3Origin                               | 88a5eaf4-2fd4-4709-b370-b4c650ea3fcf | whitelist       | {"Items": ["origin", "access-control-request-headers", "access-control-request
| Managed-Elemental-MediaTailor-PersonalizedManifests | 775133bc-15f2-49f9-abea-afb2e0bf67d2 | whitelist       | {"Items": ["origin", "access-control-request-headers", "x-forwarded-for", "acc
| Managed-UserAgentRefererHeaders                     | acba4595-bd28-49b8-b9fe-13317c0390fa | whitelist       | {"Items": ["referer", "user-agent"], "Quantity": 2}                           
| Managed-AllViewer                                   | 216adef6-5c7f-47e4-b989-5492eafa07d3 | allViewer       | <null>                                                                        
| Managed-CORS-CustomOrigin                           | 59781a5b-3903-41f3-afcb-af62929ccde1 | whitelist       | {"Items": ["origin"], "Quantity": 1}                                          
+-----------------------------------------------------+--------------------------------------+-----------------+-------------------------------------------------------------------------------
  
```
</details>

### CODE REVIEW

File: aws/table_aws_cloudfront_origin_request_policy.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+				Transform:   transform.FromField("Id", "OriginRequestPolicy.Id"),
+			},
+			{
+				Name:        "comment",
+				Description: "The comment for this origin request policy.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Comment", "OriginRequestPolicy.OriginRequestPolicyConfig.Comment"),
+			},
+			{
+				Name:        "e_tag",
```
Comment (khushboo9024) [inline aws/table_aws_cloudfront_origin_request_policy.go:47]:
```suggestion
				Name:        "etag",
```

File: aws/table_aws_cloudfront_origin_request_policy.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getCloudFrontOriginRequestPolicy,
+			},
+			{
+				Name:        "last_modified_time",
+				Description: "The date and time when the origin request policy was last modified.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("LastModifiedTime", "OriginRequestPolicy.LastModifiedTime"),
+			},
+			// We are getting  type property only in list call response
```
Comment (khushboo9024) [inline aws/table_aws_cloudfront_origin_request_policy.go:58]: Remove this `type` field.

**Comment by anisadas** [comment]
```
SETUP: tests/aws_cloudfront_origin_request_policy []

PRETEST: tests/aws_cloudfront_origin_request_policy

TEST: tests/aws_cloudfront_origin_request_policy
Running terraform

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aws_cloudfront_origin_request_policy.named_test_resource will be created
  + resource "aws_cloudfront_origin_request_policy" "named_test_resource" {
      + comment = "turbottest44793"
      + etag    = (known after apply)
      + id      = (known after apply)
      + name    = "turbottest44793"

      + cookies_config {
          + cookie_behavior = "whitelist"

          + cookies {
              + items = [
                  + "example",
                ]
            }
        }

      + headers_config {
          + header_behavior = "whitelist"

          + headers {
              + items = [
                  + "example",
                ]
            }
        }

      + query_strings_config {
          + query_string_behavior = "whitelist"

          + query_strings {
              + items = [
                  + "example",
                ]
            }
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + account_id    = "986325076436"
  + aws_partition = "aws"
  + etag          = (known after apply)
  + resource_aka  = (known after apply)
  + resource_id   = (known after apply)
  + resource_name = "turbottest44793"
aws_cloudfront_origin_request_policy.named_test_resource: Creating...
aws_cloudfront_origin_request_policy.named_test_resource: Creation complete after 3s [id=334c9674-a0cc-4ee4-b9c9-eb8639cd7f34]

Warning: Deprecated Resource

  with data.null_data_source.resource,
  on variables.tf line 43, in data "null_data_source" "resource":
  43: data "null_data_source" "resource" {

The null_data_source was historically used to construct intermediate values
to re-use elsewhere in configuration, the same can now be achieved using
locals

(and one more similar warning elsewhere)

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = "986325076436"
aws_partition = "aws"
etag = "E23ZP02F085DFQ"
resource_aka = "arn:aws:cloudfront::986325076436:origin-request-policy/334c9674-a0cc-4ee4-b9c9-eb8639cd7f34"
resource_id = "334c9674-a0cc-4ee4-b9c9-eb8639cd7f34"
resource_name = "turbottest44793"

Running SQL query: test-get-query.sql
[
  {
    "account_id": "986325076436",
    "comment": "turbottest44793",
    "etag": "E23ZP02F085DFQ",
    "id": "334c9674-a0cc-4ee4-b9c9-eb8639cd7f34",
    "title": "turbottest44793"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "account_id": "986325076436",
    "comment": "turbottest44793",
    "etag": "E23ZP02F085DFQ",
    "id": "334c9674-a0cc-4ee4-b9c9-eb8639cd7f34",
    "name": "turbottest44793",
    "title": "turbottest44793"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:cloudfront::986325076436:origin-request-policy/334c9674-a0cc-4ee4-b9c9-eb8639cd7f34"
    ],
    "title": "turbottest44793"
  }
]
✔ PASSED

POSTTEST: tests/aws_cloudfront_origin_request_policy

TEARDOWN: tests/aws_cloudfront_origin_request_policy

SUMMARY:

1/1 passed.
```

**Review by khushboo9024 — APPROVED** [review]

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**